docker/attestation-container/Dockerfile

```diff
@@ -8,15 +8,18 @@
 ARG variant=run
 
 # Build
-FROM mcr.microsoft.com/oss/go/microsoft/golang:1.19-fips-cbl-mariner2.0 AS builder-app
+FROM mcr.microsoft.com/oss/go/microsoft/golang:1.21-fips-cbl-mariner2.0 AS builder-app
 WORKDIR /usr/src/app
 
+# Set environment variables for golang fips build
+# https://github.com/microsoft/go/tree/microsoft/main/eng/doc/fips#usage-build
+ENV GOOS=linux GOEXPERIMENT=opensslcrypto CGO_ENABLED=1
```

Comment:
Might as well set this to the recommended GOEXPERIMENT=systemcrypto instead of statically picking.